src/redis/cache_impl.go

```diff
@@ -19,12 +20,26 @@ func NewRateLimiterCacheImplFromSettings(s settings.Settings, localCache *freeca
 	otherPool = NewClientImpl(srv.Scope().Scope("redis_pool"), s.RedisTls, s.RedisAuth, s.RedisType, s.RedisUrl, s.RedisPoolSize,
 		s.RedisPipelineWindow, s.RedisPipelineLimit)
 
-	return NewFixedRateLimitCacheImpl(
-		otherPool,
-		perSecondPool,
-		timeSource,
-		jitterRand,
-		expirationJitterMaxSeconds,
-		localCache,
-		s.NearLimitRatio)
+	if s.RateLimitAlgorithm == settings.FixedRateLimit {
+		return NewFixedRateLimitCacheImpl(
+			otherPool,
+			perSecondPool,
+			timeSource,
+			jitterRand,
+			expirationJitterMaxSeconds,
+			localCache,
+			s.NearLimitRatio)
+	} else if s.RateLimitAlgorithm == settings.WindowedRateLimit {
+		return NewWindowedRateLimitCacheImpl(
+			otherPool,
+			perSecondPool,
+			timeSource,
+			jitterRand,
+			expirationJitterMaxSeconds,
+			localCache,
+			s.NearLimitRatio)
+	} else {
+		logger.Fatalf("Unknown rate limit algorithm. %s\n", s.RateLimitAlgorithm)
```

Comment:
I think this is going to crash in some strange way after returning nil? Would it be better to panic here or should we have some more graceful handling somewhere?

Reply:
Throwing error and let the runner handle the error. Log level parser also trigger panic in the runner